### ISSUE TITLE
CMS Design: Enable Facility Editors to publish events to the National Outreach Calendar

### ISSUE THREAD

**Issue by wesrowe** [department-of-veterans-affairs/va.gov-cms#14788]
### User story

User stories from DaveC:

~AS AN Outreach Events editor (and only outreach events editor)~
~I WANT my event to get posted on the Outreach Events calendar by default (without any specific input from me)~
~SO THAT I don't have any new cognitive load added.~ _complicated; moved to new ticket_

AS AN Editor who can create events for a Facility
I WANT to be given the option of publishing to the Outreach Events calendar (i.e. "national" calendar) in addition to my usual options
SO THAT I can easily expose an event with broad appeal and remote accessibility to all Veterans.

AS A VAMC Editor with access to one or more VAMCs
I WANT the options to publish an event to one or more of my VAMCs AND the Outreach events calendar
SO THAT I can promote the event where I think most appropriate.

**Current Event creation form:**
https://prod.cms.va.gov/node/add/event

**Design**
[Figma file](https://www.figma.com/file/8ZT1nnu8r2twbOlEhysnNp/Events?type=design&node-id=0%3A1&mode=design&t=EsFeeJGKPh7v29n8-1)

## S91 Acceptance criteria - 8pts

- [x] Designs are created [in Figma](https://www.figma.com/file/8ZT1nnu8r2twbOlEhysnNp/Events?type=design&node-id=0%3A1&mode=design&t=EsFeeJGKPh7v29n8-1) for Event create/edit UI
  - [x] Include views for various Editors called out in user stories (Outreach-only, single-VAMC, multi-VAMC)

## S92 / SPrint 93 Acceptance criteria - 5pts
- [ ] Get review from Drupal folk (Christia, Daniel) for technical feasibility
- [ ] Get review/feedback from Amanda
- [ ] Schedule review with @wesrowe, @xiongjaneg -- Wes to partner on next steps after review (if this needs bigger review with Dave / Michelle, etc)
- [ ] Check with Blake re: whether CMS Collab Cycle review is required for this change
  - [x] update: filed [CMS CC intake ticket](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/15169)
  - [ ] Further updated CMS CC intake ticket when we have a decision on MVP – respond to [Blake's comment](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/15169#issuecomment-1714513240)

**Comment by wesrowe** [comment]
@thejordanwood, we'll be talking about this ticket in planning. I know it's last minute, but review ahead of time if you're able.
cc @aklausmeier

**Comment by wesrowe** [comment]
pulled in because Jordan had bandwidth, was waiting on something for other commitments

**Comment by thejordanwood** [comment]
**End of sprint update:**

I've made a first pass at design in [this Figma file](https://www.figma.com/file/8ZT1nnu8r2twbOlEhysnNp/Events?type=design&node-id=0%3A1&mode=design&t=EsFeeJGKPh7v29n8-1), but still need to get feedback on this.

**Comment by wesrowe** [comment]
@thejordanwood, can you help me figure out what the next step would be on this? I don't see the discussion here, but @aklausmeier and I recall your negotiating that PW might build the multi-select CMS component.

**Comment by thejordanwood** [comment]
@wesrowe An update on this, I've been working with @BlakeOrgan and @laflannery to find a component that we can bring into our CMS and use. Laura has been testing multiple components to figure out which ones are more accessible.  We haven't been able to find a component yet, but she is going to test another one tomorrow. 

I've started [a Figma file](https://www.figma.com/file/8ZT1nnu8r2twbOlEhysnNp/Events?type=design&node-id=0%3A1&mode=design&t=Gf9x88LvnQJ1x4e7-1) to show what component we're looking into and how it could help solve this problem. I'm going to provide a screenshot here too. 

<img width="679" alt="Screenshot 2023-09-07 at 5 49 28 PM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/104526584/6450ee98-859f-4d45-bdf6-7d3b52ddb407">

**Comment by wesrowe** [comment]
[Slack thread](https://dsva.slack.com/archives/C52CL1PKQ/p1694197792071149) discussion revealed:
- The multi-select dropdown would not be a quick win because no acceptable module exists
- Dave strongly prefers a quick win to deliver value
- @thejordanwood has three alternative ideas:
  - 1) One option would be to use a modal component similar the [Video list content type](https://staging.cms.va.gov/node/add/media_list_videos) that allows users to select multiple benefit hubs.
  - 2) If we're able to change the experience for editors based on their role, we could possibly get rid of the dropdown for most editors and have a checkbox list. I'm assuming most editors would see less than 10 options at a time. For top level admins we would still need a dropdown. 
  - 3) We can add a "publish to national outreach calendar" checkbox below the current dropdown.

@chri5tia @dsasser, I would like to get your thoughts on the options:
- option 1: is this viable for all Editors, whose range of options could go from 2 (single-facility editor) to hundreds (? however many Admins see)
- option 2: would it be sustainable (reasonable tech debt) to have the UI adapt its approach to different roles?
- option 3: is this as easy as it seems at my first glance?

**Comment by thejordanwood** [comment]
**End of sprint update:**

This is probably going to carry over into sprint 93. It won't take long to show whatever design solution we decide on [in Figma](https://www.figma.com/file/8ZT1nnu8r2twbOlEhysnNp/Events?type=design&node-id=0%3A1&mode=design&t=l77G2UYPdw5aquKK-1), but it'll take a bit to gather all feedback.

**Comment by jilladams** [comment]
CMS Collab cycle ticket, related: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/15169

**Comment by dsasser** [comment]
@wesrowe @thejordanwood 

> option 1: is this viable for all Editors, whose range of options could go from 2 (single-facility editor) to hundreds (? however many Admins see)
option 2: would it be sustainable (reasonable tech debt) to have the UI adapt its approach to different roles?
option 3: is this as easy as it seems at my first glance?

option 1:
We could implement this with a few configuration changes, and no custom code. This is a pattern used in m any places in the site already, so there should be low/no cognitive load for editors.

option 2:
Swapping the type of element, depending on the users' role, will be difficult to achieve, and have guaranteed tech debt. This is also not a known paradigm being used elsewhere in the CMS.

option 3:
Adding a checkbox is very easy, indeed. However, I don't know what the checkbox is actually doing, so it is hard to say if this is a viable option.

===

Re-clarifying the options:

1. One option would be to use a modal component similar the [Video list content type](https://staging.cms.va.gov/node/add/media_list_videos) that allows users to select multiple benefit hubs.
2. If we're able to change the experience for editors based on their role, we could possibly get rid of the dropdown for most editors and have a checkbox list. I'm assuming most editors would see less than 10 options at a time. For top level admins we would still need a dropdown.
3. We can add a "publish to national outreach calendar" checkbox below the current dropdown.

**Comment by davidconlon** [comment]
From what I can tell, Option 3 gets us to get the content to Veterans the fastest.

We still don't know how big a problem it is for cross-VAMC publication. 
So we can change/iterate on this later if/when we decide this is a bit enough problem to solve/get enough complaints about it.

**Comment by wesrowe** [comment]
thank you all for the discussion. MVP chosen.

**Comment by wesrowe** [comment]
@thejordanwood, I left a comment in the Figma – I want to make sure we've thought through what the interface looks like for Outreach-calendar-only Editors. We wouldn't want the checkbox and dropdown to have redundant/conflicting information. It may be a Drupal engineer question which way of handling it would be better from a tech-debt perspective.

**Comment by thejordanwood** [comment]
@chri5tia @dsasser I have a couple of questions about how the events list dropdown is handled on this content type. 

1. The current experience for this dropdown is that everyone sees all VAMCs and the national outreach calendar, regardless of their permissions. Laura [opened a ticket](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/12737) that implies this may be a defect. Is this really a defect or is it the intended experience? 
2. Ideally, editors would only be able to see the VAMCs that they're allowed to post events on. Is there a way we could make that possible?

I know I've brought this up before, but am not sure we reached a conclusion on what was possible for this.

**Comment by thejordanwood** [comment]
@wesrowe Summarizing what Daniel and I discussed in [this slack thread](https://dsva.slack.com/archives/C52CL1PKQ/p1694795111876109):

1. It's possible to make the events dropdown permissions aware, but this would take some discovery work.
2. An engineering/UI collab spike would be needed to determine how to remove the outreach calendar from the dropdown or make the "outreach editor only" view different from the "VAMC editor" view. 

If we want a quick win, @dsasser's suggestion is to move forward with just adding the checkbox for now. I'm in support of that and think we can move these additional updates into a future ticket, similar to the one you opened for [Enabling Facility Editors to publish events to multiple facilities](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/15259).

**Comment by jilladams** [comment]
The backend implementation ticket, #10089 , had both National Events calendar and Multi-VAMCs lumped into one implementation. I just broke them apart, and #15295 will track the multi-VAMC work. So to recap we've got: 

National Event calendar: 
* CMS design (here): #14788
* Implementation: #10089 

Multi-VAMCs: 
* CMS design: #15259 
* Implementation: #15295

**Comment by wesrowe** [comment]
@thejordanwood @jilladams, I agree with the quick win of the checkbox. However, I haven't seen if we have resolved what the Outreach-only Editor would see – I think it would be very confusing to offer them both 1) select the Oureach cal from the dropdown 2) check the box. Did @dsasser think we can hide one of the two options (probably the checkbox?) for this user?

**Comment by thejordanwood** [comment]
@wesrowe Right now all editors see every VAMC and the outreach calendar in the dropdown. To be able to make the dropdown permissions aware or to remove the checkbox just from the the Outreach-only Editor view would need more discovery work/a spike ticket to figure out how to do this.

**Comment by wesrowe** [comment]
@thejordanwood, I'm moving the auto-select user story to a new ticket. This can close when you're ready otherwise.
